hammer/hammer.go

```diff
@@ -182,6 +190,26 @@ func (h *Hammer) Run(ctx context.Context) {
 				if newSize > size {
 					klog.V(1).Infof("Updated checkpoint from %d to %d", size, newSize)
 				}
+				now := time.Now()
+				totalLatency := time.Duration(0)
+				queueLatency := time.Duration(0)
+				numLeaves := 0
+				for {
+					l, ok := <-h.leafSampleChan
+					if !ok {
+						break
+					}
+					if l.queuedAt.After(now) {
+						// We've caught up with everything which happened in this "epoch"
+						break
+					}
+					queueLatency += l.assignedAt.Sub(l.queuedAt)
+					totalLatency += now.Sub(l.queuedAt)
+
+					numLeaves++
+				}
+				h.integrationTime.Add(float64(totalLatency/time.Millisecond) / float64(numLeaves))
+				h.queueTime.Add(float64(queueLatency/time.Millisecond) / float64(numLeaves))
```

Comment:
How about a separate goroutine for this? I don't think it needs to have a shared fate with the checkpoint updating, and separating the concerns is cleaner.

Reply:
Yeah, good call - done.

Maybe in the future there might be a case for some sort of channel/event based thing to happen on new checkpoints, but for now I've just added another ticker-based loop.